magic6/magic6.cpp

```diff
@@ -0,0 +1,83 @@
+#include <fstream>
+
+using namespace std;
+
+ifstream input("magic6.in");
+ofstream output("magic6.out");
+
+char character;
+int n;
+int frequencyVector[10] = {};
+
+void verifyIfItsIntegerOrChar()
+{
+	if (character > 47 && character < 58)
+	{
+		frequencyVector[character - '0'] = 1;
+	}
+	else
+	{
+		character = (char)(character - 32);
+		output << character;
+	}
+}
+
+void writeNumbers()
+{
+	int poz = -1;
+	int i = 1;
+
+	if (frequencyVector[0] == 1)
+	{
+
+		while (poz < 0)
+		{
+			if (frequencyVector[i] == 1)
+			{
+				poz = i;
+			}
+			else
+			{
+				i++;
+			}
+		}
+		output << poz << 0;
```

Comment:
you should wrap up in a function

Reply:
it's not working, if i put it in a function with 2 parameters(one for poz and second for index) and i call it in my writeNumbers() function, output is not showing the correct answer

Reply:
// void verifyFirstDigitAfter0(int position, int index)
// {
// 	while (position < 0)
// 		{
// 			if (frequencyVector[index] == 1)
// 			{
// 				position = index;
// 			}
// 			else
// 			{
// 				index++;
// 			}
// 		}
// 		output << position << 0;
// }

Reply:
a parameter which will be modified in the scope should be a referenced one (&)
remember that!!! we discussed about that

Reply:
you don't need the index parameter

Reply:
because index is used only in this scope